pkg/apis/sparkscheduler/v1beta1/conversion_resource_reservation_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2019 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These tests test conversion between v1beta1 and v1beta2 of the resource reservation CRD.
+package v1beta1
```

Comment:
Can you please also add tests for cases such as adding an executor reservation in v1 that didn't exist in the marshalled v2 annotation and similarly removing a reservation?